src/mlir/dialect/VeronaTypes.cc

```diff
@@ -316,8 +317,11 @@ namespace mlir::verona
 
   ClassType::FieldsRef ClassType::getFields() const
   {
-    assert(getImpl()->isInitialized);
-    return getImpl()->fields;
+    // We may not have a full declaration available
+    if (getImpl()->isInitialized)
+      return getImpl()->fields;
+    else
+      return ClassType::FieldsRef();
```

Comment:
How does a caller differentiate between not initialised yet and has no fields?

Reply:
It does not. This is one of the things that will need to be changed later.

Right now, we're lowering "classes" that have no fields declared (like U32) mostly because we're not declaring it at all. This is the main reason why we need this change.

With time, and only after we implement modules/includes, we'll be able to assert on initialisation. Without this, no class type can be printed back.

Reply:
I've added a TODO to remind us that this needs to be an assert in a complete compiler.